packages/core/src/Page/PageVueServerRenderer.js

```diff
@@ -0,0 +1,132 @@
+const path = require('path');
+const fs = require('fs-extra');
+
+const Vue = require('vue');
+const VueCompiler = require('vue-template-compiler');
+const { renderToString } = require('vue-server-renderer').createRenderer();
+
+let bundle = require('@markbind/core-web/dist/js/markbindvue.min');
+
+const logger = require('../utils/logger');
+
+/*
+ * Suppress irrelevant Vue warnings that comes up during SSR.
+ * These warnings are expected false-positives that do not affect the outcome of SSR.
+ *
+ * Following are some of the warnings in general form.
+ * E.g. "Property or method is not defined on the instanced but referenced during render."
+ * E.g. "'key' is a reserved attribute and cannot be used as component prop."
+ * E.g. "Injection not found"
+ * E.g. "Avoid mutating a prop directly since the value will be overwritten whenever the parent
+ * component re-renders. Instead, use a data or computed property based on the prop's value."
+ * E.g. "Invalid prop: type check failed for prop. Expected Number, got String."
+ */
+Vue.config.silent = true;
```

Comment:
> hmm... not sure if this is a good idea - future new warnings will be silently ignored, even if they are important.

Hmm that's true. How about we set it to false during `-d` mode? 

> should we investigate and fix each of these errors in separate commits?

Actually these warnings also appear on the browser. But we use production Vue in the browser thus these warnings do not appear in the browser console (can try switching to development Vue using their CDN to see these warnings). The reason these warnings appear is because the variables are "used" before we have defined them. But these variables will eventually be defined in the browser, so there is no problem. 

In our SSR, even though these variables will never be defined, since we are merely rendering the app in HTML string, I believe there will be no issues posed to the outcome of the SSR. One possible way to resolve these warnings is to pass in / declare these variables when we are creating the Vue instance for each page. 

We can do something like below. I managed to "fix" most of the errors by passing in "stubbed" variables while I was experimenting with SSR. 

```
const VueAppPage = new Vue({
    template: `<div id="app">${content}</div>`,
    data: {
       searchData: [],
       popoverInnerGetters: {},
       tooltipInnerContentGetter: {},
       fragment: '',
       questions: [],
     },
     props: {
       threshold: Number(0.5),
     },
     methods: {
       searchCallback: () => {},
     },
     provide: {
       hasParentDropdown: true,
       questions: [],
       gotoNextQuestion: true,
     },
  });
```

This issue is a little tricky in my opinion. On one hand, we don't want to disable the warnings in case there are important warnings that contributors may miss out. On the other hand, we don't really want to see these "false-positive" warnings. One way to fix this is to pass in "stubbed" variables but I'm not sure if it's the best way. What do you think?

Reply:
> One way to fix this is to pass in "stubbed" variables but I'm not sure if it's the best way.

That's ok, as far as I can see there are only a couple (`searchData`, `searchCallback`, ...). We can extract these already present common options in `core-web/index` to a common file and import from there as well to remove duplication.

i.e. a common `App` factory, per https://ssr.vuejs.org/guide/structure.html

Reply:
Yup, I agree with this. I've made a new file called `vueCommonAppFactory` to ensure we have a universal Vue application. Let me know what you think about it :-)